### PR TITLE
Set contact colours on uids

### LINKED ISSUE #149: Reuse colors from the contact app
# Current behavior

ocsms seems to generate new colors for each conversation which may be different from the ones used in contacts app

# Expected behavior

ocsms should check if the contact exists in the contacts app and reuse the color from there

**When** there is a contact in the contacts app that matches the entry in ocsms
**Then** the color from the contacts app should be used

**When** there is no contact that can be associated
**Then** you can define your own color *or* use a static grey color (because it could simplify your code if you don't need color generators at all - assuming most people who use ocsms will host their contacts at nextcloud, too)

# Benefit

1. Less confusion because the user can assign **one** color to **one** contact (1:1 relation)
2. Better colors - sorry, but contacts app is simply doing it better (nicer colors, better contrast, better readability)

### PR DESCRIPTION
**Pull request by skjnldsv**
Fix #149

### CODE REVIEW

File: lib/contactcache.php [hunk 1]
```diff
@@ -99,6 +109,10 @@ private function loadContacts() {
 					$photoURL = preg_replace("#^VALUE=uri:#","",$r["PHOTO"], 1);
 					$this->contactPhotos[$r["FN"]] = $photoURL;
 				}
+				
+				if (isset ($r["UID"])) {
```
Comment (skjnldsv) [inline lib/contactcache.php]: This shouldn't happen, but just in case.

**Comment by skjnldsv** [comment]
There is an issue with the initial load when the contact list isn't loaded yet.
https://github.com/nerzhul/ocsms/blob/master/js/public/app.js#L504

uid isn't set. @nerzhul Do you mind checking it and see what you can do to be sure the contact list is  loaded before rendering?

**Comment by nerzhul** [comment]
@skjnldsv thanks for this interesting fix. Maybe we can init UID with 0 value, it will be recalculated later, what do you think about that ?

**Comment by skjnldsv** [comment]
Not the best option. Why don't you wait for the contact app to finish loading?
Seems to be the best way to do so :)

**Comment by nerzhul** [comment]
You are right, we should wait contact loading, and then trigger the conversation load or recalculate colors after contact loading
I will merge this fix first and then we will fix the other issue

**Pull request merged**